### ISSUE TITLE
Unable to use plugins that reference NewtonSoft.Json 11.0.2

### ISSUE THREAD

**Issue by JeremyTCD** [dotnet/docfx#3471]
**DocFX Version Used**:  v2.39.2

**Template used**: n/a

**Steps to Reproduce**:

1. Create a simple post processor that references NewtonSoft.Json 11.0.2.
2. Use post processor.

**Expected Behavior**:
DocFX runs as per normal.

**Actual Behavior**:
The following exception is thrown:
```
System.IO.FileLoadException: Could not load file or assembly 'Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
```
This seems to occur because DocFX uses NewtonSoft.Json 9.0.1.  so in my clone of DocFX, I added a reference to NewtonSoft.Json 11.0.2  in the DocFX project. Upon building the project, the following binding redirect was automatically added to docfx.exe.config in the output directory:  
``` xml
<dependentAssembly>
    <assemblyIdentity name="Newtonsoft.Json" publicKeyToken="30ad4fe6b2a6aeed" culture="neutral" />
    <bindingRedirect oldVersion="0.0.0.0-11.0.0.0" newVersion="11.0.0.0" />
</dependentAssembly>
```
However, the binding redirect did not work. Now the following exception occurs:
```
System.IO.FileLoadException: Could not load file or assembly 'Newtonsoft.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
```
The requesting assembly causing the exception is now `Microsoft.DocAsCode.Plugins`.

I'm not sure if this is the correct place for this issue, and would be more than happy to move it elsewhere if necessary. Just thought you all might know what to do if plugins reference different versions of assemblies that are already loaded.

tl:dr - Attempted to use a plugin that references a newer NewtonSoft.Json version, could not get DocFX to work even with binding redirects. Any input highly appreciated.

**Comment by JeremyTCD** [comment]
Just took a look at V3. On second thought, not much point trying to fix my issue since it won't happen with V3's plugin system. For anyone else who encounters the same issue, until V3 is released, i'm using temporary assemblies that all reference the same NewtonSoft.Json version.

**Comment by JeremyTCD** [comment]
Turns out that DocFX allows for a custom `.config` file for the plugin's `AppDomain`: 

https://github.com/dotnet/docfx/blob/655d99caf06acaa9c97cee2b51deb1f5ad338e1d/src/docfx/SubCommands/BuildCommand.cs#L498-L508

My binding redirects did not work because I wrongly assumed that the default app.config was used. Works fine now.
